extensions/arc/deployment/src/test/java/io/quarkus/arc/test/unproxyable/PrivateNoArgsConstructorDoNotChangeFlagTest.java

```diff
@@ -0,0 +1,48 @@
+package io.quarkus.arc.test.unproxyable;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.spi.DeploymentException;
+
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.arc.Unremovable;
+import io.quarkus.test.QuarkusUnitTest;
+
+public class PrivateNoArgsConstructorDoNotChangeFlagTest {
```

Comment:
These tests are great! I think that having a extra test where the bean extends some class that doesn't have a no-args constructor would be beneficial - to ensure that a build error is encountered

Reply:
There is/was `FailedToAddMissingNoargsConstructorTest`... is that what you mean?

Reply:
Oh yes! That is exactly what I meant. I missed that one